test/e2e/cron_test.go

```diff
@@ -327,23 +324,17 @@ spec:
           args: ["exit", "1"]`).
 			When().
 			CreateCronWorkflow().
-			Wait(2*time.Minute+35*time.Second).
+			Wait(2*time.Minute+20*time.Second).
 			Then().
 			ExpectWorkflowList(listOptions, func(t *testing.T, wfList *wfv1.WorkflowList) {
-				assert.LessOrEqual(t, 2, len(wfList.Items))
-				assert.True(t, wfList.Items[len(wfList.Items)-1].Status.FinishedAt.Time.After(time.Now().Add(-1*time.Minute)))
+				assert.Equal(t, 1, len(wfList.Items))
+				assert.True(t, wfList.Items[0].Status.FinishedAt.Time.After(time.Now().Add(-1*time.Minute)))
 			})
 	})
 }
 
 func wfInformerListOptionsFunc(options *v1.ListOptions, cronWfName string) {
-	options.FieldSelector = fields.Everything().String()
-	isCronWorkflowChildReq, err := labels.NewRequirement(common.LabelKeyCronWorkflow, selection.Equals, []string{cronWfName})
-	if err != nil {
-		panic(err)
-	}
-	labelSelector := labels.NewSelector().Add(*isCronWorkflowChildReq)
-	options.LabelSelector = labelSelector.String()
+	options.LabelSelector = common.LabelKeyCronWorkflow + "=" + cronWfName
```

Comment:
I was wrong about this change, I think you may have been correct

Reply:
Worth a try anyway :)